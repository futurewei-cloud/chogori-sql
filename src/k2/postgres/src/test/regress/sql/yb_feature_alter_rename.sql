
--
-- ALTER RENAME
-- rename column, table, database
--

create database test_rename;
create database test_rename1;
\c test_rename

create table foo(a int primary key, b int);
insert into foo (a, b) values (1, 2);
\d
\d foo

alter table foo rename column b to c;
select a, b from foo; -- fail
select a, c from foo;
insert into foo (a, b) values (2, 3); -- fail
insert into foo (a, c) values (3, 4);
\d foo

alter table foo rename to bar;
select * from foo; -- fail
select * from bar;
\d

\c test_rename1;
alter database test_rename rename to test_rename2;
alter database test_rename2 rename to postgres; -- fail
\l
