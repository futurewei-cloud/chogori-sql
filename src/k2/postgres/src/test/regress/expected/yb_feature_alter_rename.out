--
-- ALTER RENAME
-- rename column, table, database
--
create database test_rename;
create database test_rename1;
\c test_rename
create table foo(a int primary key, b int);
insert into foo (a, b) values (1, 2);
\d
        List of relations
 Schema | Name | Type  |  Owner   
--------+------+-------+----------
 public | foo  | table | yugabyte
(1 row)

\d foo
                Table "public.foo"
 Column |  Type   | Collation | Nullable | Default 
--------+---------+-----------+----------+---------
 a      | integer |           | not null | 
 b      | integer |           |          | 
Indexes:
    "foo_pkey" PRIMARY KEY, lsm (a HASH)

alter table foo rename column b to c;
select a, b from foo; -- fail
ERROR:  column "b" does not exist
LINE 1: select a, b from foo;
                  ^
select a, c from foo;
 a | c 
---+---
 1 | 2
(1 row)

insert into foo (a, b) values (2, 3); -- fail
ERROR:  column "b" of relation "foo" does not exist
LINE 1: insert into foo (a, b) values (2, 3);
                            ^
insert into foo (a, c) values (3, 4);
\d foo
                Table "public.foo"
 Column |  Type   | Collation | Nullable | Default 
--------+---------+-----------+----------+---------
 a      | integer |           | not null | 
 c      | integer |           |          | 
Indexes:
    "foo_pkey" PRIMARY KEY, lsm (a HASH)

alter table foo rename to bar;
select * from foo; -- fail
ERROR:  relation "foo" does not exist
LINE 1: select * from foo;
                      ^
select * from bar;
 a | c 
---+---
 1 | 2
 3 | 4
(2 rows)

\d
        List of relations
 Schema | Name | Type  |  Owner   
--------+------+-------+----------
 public | bar  | table | yugabyte
(1 row)

\c test_rename1;
alter database test_rename rename to test_rename2;
alter database test_rename2 rename to postgres; -- fail
ERROR:  database "postgres" already exists
\l
                                   List of databases
      Name       |  Owner   | Encoding | Collate |    Ctype    |   Access privileges   
-----------------+----------+----------+---------+-------------+-----------------------
 postgres        | postgres | UTF8     | C       | en_US.UTF-8 | 
 system_platform | postgres | UTF8     | C       | en_US.UTF-8 | 
 template0       | postgres | UTF8     | C       | en_US.UTF-8 | =c/postgres          +
                 |          |          |         |             | postgres=CTc/postgres
 template1       | postgres | UTF8     | C       | en_US.UTF-8 | =c/postgres          +
                 |          |          |         |             | postgres=CTc/postgres
 test_rename1    | yugabyte | UTF8     | C       | en_US.UTF-8 | 
 test_rename2    | yugabyte | UTF8     | C       | en_US.UTF-8 | 
 yugabyte        | postgres | UTF8     | C       | en_US.UTF-8 | 
(7 rows)

